therapy/etl/rxnorm.py

```diff
@@ -0,0 +1,441 @@
+            else:
+                self._add_term(params, term, 'aliases')
+        elif term_type in TRADE_NAMES:
+            self._add_term(params, term, 'trade_names')
+
+        if source == 'RXNORM':
+            if term_type == 'SBDF':
+                brand = term.split('[')[-1].split(']')[0]
+                ingredient_strength = term.replace(f"[{brand}]", '')
+                for df in drug_forms:
+                    if df in ingredient_strength:
+                        ingredient = \
+                            ingredient_strength.replace(df, '').strip()
+                        self._add_term(sbdfs, brand, ingredient.lower())
+                        break
+
+        if source == 'MSH':
+            if term_type == 'MH':
+                # Get ID for accessing precise ingredient
+                params['PIN'] = row[13]
+            elif term_type == 'PEP':
+                self._add_term(precise_ingredient, term, row[13])
+
+    def _add_term(self, params, term, label_type):
+        """Add a single term to a therapy record in an associated field.
+
+        :param dict params: A transformed therapy record.
+        :param str term: The term to be added
+        :param str label_type: The type of term
+        """
+        if label_type in params and params[label_type]:
+            if term not in params[label_type]:
+                params[label_type].append(term)
+        else:
+            params[label_type] = [term]
+
+    def _add_other_ids_xrefs(self, params, row):
+        """Add other identifier or xref to therapy.
+
+        :param dict params: A transformed therapy record.
+        :param list row: A row in the RxNorm data file.
+        """
+        if row[11]:
+            other_id_xref = row[11].upper()
+            if other_id_xref in self._other_id_srcs:
+                source_id =\
+                    f"{NamespacePrefix[other_id_xref].value}:{row[13]}"
+                if source_id != params['concept_id']:
+                    # Sometimes concept_id is included in the source field
+                    self._add_term(params, source_id, 'other_identifiers')
+            elif other_id_xref in self._xref_srcs:
+                source_id = f"{NamespacePrefix[other_id_xref].value}:{row[13]}"
+                self._add_term(params, source_id, 'xrefs')
+            else:
+                logger.info(f"{other_id_xref} not in NameSpacePrefix.")
+
+    def _add_meta(self):
+        """Add RxNorm metadata."""
+        meta = Meta(data_license='UMLS Metathesaurus',
+                    data_license_url='https://www.nlm.nih.gov/research/umls/'
+                                     'rxnorm/docs/termsofservice.html',
+                    version=self._version,
+                    data_url=self._data_url,
+                    rdp_url=None,
+                    data_license_attributes={
+                        'non_commercial': False,
+                        'share_alike': False,
+                        'attribution': False
```

Comment:
ChemIDPlus has `attribution = True`. So this may be incorrect.

Reply:
I changed ChemIDplus to `False` for now -- we can update once Alex has reviewed the license terms